interval/interval.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package interval provides a struct representing a time interval.
+package interval
+
+import "time"
+
+// Interval represents the interval [Start, End).
+type Interval struct {
```

Comment:
We have a stuttery name: `interval.Interval`.

Ideally this would be `package time` giving us `time.Interval` but I can see the amount of confusion that could cause and the inevitable package aliasing all your code will end up doing.

Could `timeutil` be justified?  `timeutil.Interval`

Reply:
It does stutter, but I feel like I've seen this particular brand of stutter before - specifically when a package contains one main type and a bunch of stuff to do with that type.  Because if, for example, I had a constructor function in this package (which admittedly I don't), it'd be something like 
```
func New(start, end time.Time) Interval {...}
```
and you'd call it using 
```
i := interval.New(start, end)
```
which is pretty typical right?

The time package itself actually stutters this way - time.Time and all that jazz.

Reply:
OK.... I'm not convinced you should add a constructor function for a type that's as simple as Interval.  But let's leave it as it is for now.